### DISCUSSION TITLE
[Support]: How to directly access live stream of camera

### DISCUSSION THREAD

**Discussion by nelsongraca** [blakeblackshear/frigate#13623] [General]
Hello, I have a USB camera connected to my frigate host, frigate side all working ok.

But this camera is for watching over my 3D printer and I would like to integrate it on the web interface of the printer, it allows me either mjpeg stream or go2rtc (depending on the UI used and I actually need both methods). Question is how can I access the camera streams directly so the UIs I'm using can consume it.

thanks

**Comment by hawkeye217** [comment]
If you're using go2rtc with Frigate, just make sure port 8554 is exposed in your docker compose and access the rtsp streams via Frigate's internal go2rtc directly, eg: `rtsp://frigate_ip:8554/camera_name`

**Reply by nelsongraca** [reply]
Thank you for the reply but rtsp isn't supported by the frontends I'm using, that's why I asked for MJPEG or go2rtc, I imagine if the web ui can use them there has to be a way to directly connect to them

**Reply by hawkeye217** [reply]
You can use go2rtc to transcode your cameras to an mjpeg stream.  https://github.com/AlexxIT/go2rtc?tab=readme-ov-file#module-mjpeg
